libp2p/netMessenger.go

```diff
@@ -208,10 +207,14 @@ func constructNode(
 		return nil, err
 	}
 
+	p2pSignerInstance, err := crypto.NewP2PSigner(p2pPrivateKey)
+	if err != nil {
+		cancelFunc()
```

Comment:
why not move L203 on L215 and delete `cancelFunc()` here and on L206?

Reply:
great suggestion. Done